src/Plugin/facets/facet_source/ListFacetSource.php

```diff
@@ -149,21 +149,16 @@ public function getDataDefinition($field_name) {
   public function fillFacetsWithResults(array $facets) {
     $plugin_definition_id = $this->getPluginDefinition()['id'];
     $results = $this->searchApiQueryHelper->getResults($plugin_definition_id);
-    if ($results instanceof ResultSetInterface && $results->getResultCount()) {
-      $facet_results = $results->getExtraData('search_api_facets');
-      // Loop over each facet and execute the build method from the given query
-      // type.
-      foreach ($facets as $facet) {
-        if (isset($facet_results[$facet->getFieldIdentifier()])) {
-          $configuration = [
-            'query' => NULL,
-            'facet' => $facet,
-            'results' => $facet_results[$facet->getFieldIdentifier()],
-          ];
-          $query_type = $this->queryTypePluginManager->createInstance($facet->getQueryType(), $configuration);
-          $query_type->build();
-        }
-      }
+
+    $facet_results = $results instanceof ResultSetInterface ? $results->getExtraData('search_api_facets') : [];
```

Comment:
Calling the query type plugins regardless of whether we have results. This is to give them the opportunity to create results based on active items.